fl4health/datasets/skin_cancer/load_data.py

```diff
@@ -0,0 +1,204 @@
+# The following code is adapted from the medical_federated GitHub repository by Seongjun Yang et al.
+# Paper: https://arxiv.org/abs/2207.03075
+# Code: https://github.com/wns823/medical_federated.git
+# - medical_federated/skin_cancer_federated/split_dataset.py
+# - medical_federated/skin_cancer_federated/skin_cancer_datasets.py
+
+import json
+import random
+from concurrent.futures import ThreadPoolExecutor
+from logging import ERROR, INFO
+from pathlib import Path
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+import torch
+import torchvision.transforms as transforms
+from flwr.common.logger import log
+from PIL import Image
+from torch.utils.data import DataLoader
+
+from fl4health.utils.dataset import TensorDataset
+from fl4health.utils.dataset_converter import DatasetConverter
+from fl4health.utils.sampler import LabelBasedSampler
+
+
+def construct_skin_cancer_tensor_dataset(
+    data: List[Dict[str, Any]], transform: Optional[Callable] = None, num_workers: int = 8
+) -> TensorDataset:
+    def load_image(item: Dict[str, Any]) -> Tuple[torch.Tensor, int]:
+        image_path = item["img_path"]
+        image = Image.open(image_path).convert("RGB")
+        if transform:
+            image = transform(image)
+        else:
+            # Default transformation if none provided
+            image = transforms.ToTensor()(image)
+        assert isinstance(image, torch.Tensor), f"Image at {image_path} is not a Tensor"
+        target = int(torch.tensor(item["extended_labels"]).argmax().item())
+        return image, target
+
+    with ThreadPoolExecutor(max_workers=num_workers) as executor:
+        results = list(executor.map(load_image, data))
+
+    data_list, targets_list = zip(*results)
+    data_tensor = torch.stack(list(data_list))
+    targets_tensor = torch.tensor(list(targets_list))
+
+    return TensorDataset(data_tensor, targets_tensor)
+
+
+def load_skin_cancer_data(
+    data_dir: Path,
+    dataset_name: str,
+    batch_size: int,
+    sampler: Optional[LabelBasedSampler] = None,
+    train_transform: Union[None, Callable] = None,
+    val_transform: Union[None, Callable] = None,
+    dataset_converter: Optional[DatasetConverter] = None,
+    seed: int = 0,
```

Comment:
Rather than having a default seed, I think we want to make this optional and default to None. I think the expected behavior from a user would be that, if I don't set a seed, then things like splitting wouldn't be deterministic.

Reply:
changed it to Optional!